code/game/machinery/harvester.dm

```diff
@@ -0,0 +1,159 @@
+/obj/machinery/harvester
+	name = "organ harvester"
+	desc = "An advanced machine used for harvesting organs and limbs from the deceased."
+	density = TRUE
+	icon = 'icons/obj/machines/harvester.dmi'
+	icon_state = "harvester"
+	verb_say = "states"
+	state_open = FALSE
+	idle_power_usage = 50
+	circuit = /obj/item/circuitboard/machine/harvester
+	light_color = LIGHT_COLOR_BLUE
+	var/interval = 20
+	var/harvesting = FALSE
+	var/list/operation_order = list() //Order of wich we harvest limbs.
+	var/allow_clothing = FALSE
+	var/allow_living = FALSE
+
+/obj/machinery/harvester/Initialize()
+	. = ..()
+	if(prob(5))
+		name = "auto-autopsy"
+
+/obj/machinery/harvester/RefreshParts()
+	interval = 0
+	var/max_time = 40
+	for(var/obj/item/stock_parts/micro_laser/L in component_parts)
+		max_time -= L.rating
+	if(max_time <= 0)
+		max_time = 1 //ungodly, but whatever
+	interval = max_time
+
+/obj/machinery/harvester/update_icon(warming_up)
+	if(warming_up)
+		icon_state = initial(icon_state)+"-charging"
+		return
+	if(state_open)
+		icon_state = initial(icon_state)+"-open"
+	else if(harvesting)
+		icon_state = initial(icon_state)+"-active"
+	else
+		icon_state = initial(icon_state)
+
+/obj/machinery/harvester/open_machine(drop = TRUE)
+	. = ..()
+	harvesting = FALSE
+
+/obj/machinery/harvester/attack_hand(mob/user)
+	if(state_open)
+		close_machine()
+	else if(!harvesting)
+		open_machine()
+
+/obj/machinery/harvester/AltClick(mob/user)
+	if(harvesting || !user || !isliving(user) || state_open)
+		return
+	if(can_harvest())
+		start_harvest()
+
+/obj/machinery/harvester/proc/can_harvest()
+	if(!powered(EQUIP) || state_open || !occupant || !iscarbon(occupant))
+		return
+	var/mob/living/carbon/C = occupant
+	if(!allow_clothing)
+		for(var/obj/item/I in C.held_items + C.get_equipped_items())
+			if(!(I.item_flags & NODROP))
+				say("Subject may not have abiotic items on.")
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 30, 1)
+				return
+	if(!(MOB_ORGANIC in C.mob_biotypes))
+		say("Subject is not organic.")
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 30, 1)
+		return
+	if(!allow_living && (C.stat != DEAD && !C.has_trait(TRAIT_FAKEDEATH)))     //I mean, the machines scanners arent advanced enough to tell you're alive
+		say("Subject is still alive.")
```

Comment:
Literally ling test?

Reply:
it checks if they have the FAKEDEATH trait

Reply:
so it is

Reply:
wait C.stat is unconscious for lings if they are in "fake death" arent they? fuck thats an oversight

Reply:
wait C.stat is unconscious for lings if they are in "fake death" arent they? fuck thats an oversight

Reply:
they have the regen organs ability 100% of the time just don't include the check ;)

Reply:
but can they outregen the machine?

Reply:
lings will count as harvestable so it's no ling test, when it's done they just can just regrow the organs and leave

Reply:
it gibs when it's done

Reply:
maybe it could leave the organless corpse behind after harvesting, no reason to make it gib